docs/ECLLanguageReference/ECLR_mods/Basics-Constants.xml

```diff
@@ -152,4 +152,102 @@ MyReal1 := 10.0;   // value of MyReal1 is the REAL value 10.0
 MyReal2 := 1.0e1;  // value of MyReal2 is the REAL value 10.0
 </programlisting>
   </sect2>
+
+  <sect2 id="CompileTimeConstants" role="brk">
+    <title>Compile Time Constants</title>
+
+    <para>The following system constants <indexterm>
+        <primary>constants</primary>
+      </indexterm><indexterm>
+        <primary>system constants</primary>
+      </indexterm>are available at compile time. These can be useful in
+    creating conditional code.</para>
+
+    <para><informaltable colsep="1" frame="all" rowsep="1">
+        <tgroup cols="2">
+          <colspec colwidth="150pt" />
+
+          <colspec />
+
+          <tbody>
+            <row>
+              <entry><emphasis>_ECL_VERSION_<indexterm>
+                  <primary>_ECL_VERSION_</primary>
+                </indexterm></emphasis></entry>
+
+              <entry>A STRING containing the value of the platform version.
+              For example, '6.4.0'</entry>
+            </row>
+
+            <row>
+              <entry><emphasis>_ECL_VERSION_MAJOR_ <indexterm>
+                  <primary>_ECL_VERSION_MAJOR_</primary>
+                </indexterm></emphasis></entry>
+
+              <entry>An INTEGER containing the value of the major portion of
+              the platform version. For example, '6'</entry>
+            </row>
+
+            <row>
+              <entry><emphasis>_ECL_VERSION_MINOR_<indexterm>
+                  <primary>_ECL_VERSION_MINOR_</primary>
+                </indexterm></emphasis></entry>
+
+              <entry>An INTEGER containing the value of the minor portion of
+              the platform version. For example, '4'</entry>
+            </row>
+
+            <row>
+              <entry><emphasis>_ECL_LEGACY_MODE_<indexterm>
+                  <primary>_ECL_LEGACY_MODE_</primary>
+                </indexterm></emphasis></entry>
+
+              <entry>A BOOLEAN value indicating if it is being compiled with
+              legacy IMPORT semantics.</entry>
+            </row>
+
+            <row>
+              <entry><emphasis>_OS_<indexterm>
+                  <primary>_OS_</primary>
+                </indexterm></emphasis></entry>
+
+              <entry>A STRING indicating the operating system to which it is
+              being compiled. (Available values are: 'windows' or
+              'linux')</entry>
+            </row>
+
+            <row>
+              <entry><emphasis>_STAND_ALONE_<indexterm>
+                  <primary>_STAND_ALONE_</primary>
+                </indexterm></emphasis></entry>
+
+              <entry>A BOOLEAN value indicating if it is being compiled to a
+              stand-alone executable.</entry>
+            </row>
+          </tbody>
+        </tgroup>
+      </informaltable>Example:</para>
+
+    <para><programlisting>IMPORT STD;
+  STRING14 fGetDateTimeString() :=
+  #IF(__ECL_VERSION__ &gt;= '5.2')
```

Comment:
I seem to have lost a comment on this line.  This doesn't work once the major version goes above 9.  Should be

(__ECL_VERSION_MAJOR__ > 5) or ((__ECL_VERSION_MAJOR__ = 5) AND (__ECL_VERSION_MINOR__ >= 2))

(untested)